core/src/main/java/org/apache/calcite/adapter/enumerable/RexToLixTranslator.java

```diff
@@ -341,6 +344,16 @@ private Expression converterToDate(RelDataType sourceType, Expression operand) {
           BuiltInMethod.TIMESTAMP_WITH_LOCAL_TIME_ZONE_TO_DATE.method,
           operand,
           Expressions.call(BuiltInMethod.TIME_ZONE.method, root)));
+      break;
+    // SQL 2011 Part2 Section 4.6 General Rules 13.d
+    case TIMESTAMP_WITH_TIME_ZONE:
+      convert = Expressions.convert_(
+        Expressions.call(BuiltInMethod.FLOOR_DIV.method,
```

Comment:
implement a `timestampWithoutTimeZoneToDate` and call it here?

Reply:
Not needed!
1) Section 6.13 General Rules 13.d shows this can be computed as "CAST ( CAST ( VE AS TIMESTAMP WITHOUT TIME ZONE ) AS DATE )"
2) My underlying TimestampWithTimeZone holds the same long value with Timestamp type, current implementation is more simple

Reply:
Yes. By reading the SQL standard I got the idea of the conversion path.

This suggestion is more from readability perspective. If you implements one more function that then code readers will know the equivalent implementation is  `CAST ( CAST ( VE AS TIMESTAMP WITHOUT TIME ZONE ) AS DATE )`.

I had to read SQL standard cause I wasn't sure why there is a "magic" operation after casting to `TIMESTAMP WITHOUT TIME ZONE`.

Or maybe you can add a comment instead to say the implementation here is `CAST ( CAST ( VE AS TIMESTAMP WITHOUT TIME ZONE ) AS DATE )`